test/src/module/integration/allTest.c

```diff
@@ -381,6 +381,11 @@ testRun(void)
             TEST_HOST_SQL_ONE_STR_Z(pg1, "select message from status", TEST_STATUS_TIME);
         }
 
+        // Store the time where the entire timeline captured in xidTimeline above exists so it can be used in a later test. We need
+        // to capture this here (instead of above) to be sure that all the timeline WAL has been archived.
+        const char *const xidTime = strZ(pckReadStrP(hrnHostSqlValue(pg1, "select current_timestamp::text")));
```

Comment:
What guarantees the TLI completion, just that we're using the same pg connection so it's guaranteed to be serial?

Reply:
Yes, since we are on a new timeline here the old one should be archived. If something goes wrong with that then the later test will fail when it tries to follow that timeline. Since this is a test I'm not too worried about the that happening. What we need there is a time where we are pretty sure the prior timeline has been archived in its entirety.